gitlab_integration/README.md

```diff
@@ -1,16 +1,34 @@
 # Gitlab integration research
 
+## Project integrations
+
+It allow you to integrate GitLab with other applications. They are a bit like plugins in that they allow a lot of freedom in adding functionality to GitLab.
+
+Just like Github Apps but with some limited functionality. This will be a very easy way for any projects to use the Packit service (eg in https://gitlab.gnome.org)
+
+So just by clicking one button, packit service will be active for the project - 
+
+![packit service example](./gitlab-packit.png)
+
+The above method is much easier for the user rather then creating a webhook manually.
+
+![Webhook](./webhook.png)
+
 ## Pros and Cons of having Packit as a Project Service in Gitlab
 
-A integration / project service in gitlab works in a way that the service does not have the access to the repository and can only work as a webhook to report changes based on permissions given (issues, merge requests etc) 
+A integration / project service in gitlab works in a way that the service does not have the access to the repository and can only work as a webhook to report changes based on permissions given (issues, merge requests etc).
 
 ### Pros 
 
 - Users dont have to setup webhook for each project.
-- Instead of tweaking the config file for packit you can have a settings page for the service.
+- In the later stages we can setup a settings page for the packit-service in gitlab which will act as a config for the service instead of tweeking packit config file in the repository.
 
 ### Cons
 
-- Users still have to define auth for hooks (Packit).
-- Users will still have to give access to a packit service user for getting the build updates.
+- Users will still have to give access to a packit service "user" for getting the build updates.
+ - This introduces a new programming language "Ruby" that we will need to work on for maintaining the service in gitlab.
+
+For adding the service to gitlab we will have to contribute to the https://gitlab.com/gitlab-org/gitlab which has to be approved by the admins (Currently looking more into it). 
+
+All the events created by integration / webhooks are the same so we can even take this up later.
```

Comment:
Relevant point!

In both version, how can the user specify events that will be sent to us?

Reply:
I heard that there is a way to setup webhooks globally (automatically) for all projects of the instance. (Can you please try to confirm that? It can help us a lot in case of Fedora-friendly instances like gitlab.gnome.org/.)

But that's only a shortcut and not doable for instances like gitlab.com.

Reply:
>In both versions, how can the user specify events that will be sent to us?

As you can see https://github.com/shreyaspapi/research/raw/master/gitlab_integration/gitlab-packit.png Users have trigger options for Push, Issue, Confidential Issue, Merge Request. Only if the option is ticked the events will be sent to us.

Reply:
> I heard that there is a way to setup webhooks globally (automatically) for all projects of the instance. (Can you please try to confirm that? It can help us a lot in case of Fedora-friendly instances like gitlab.gnome.org/.)
> 
> But that's only a shortcut and not doable for instances like gitlab.com.

Even if we set up webhook globally for gnome will they be comfortable for sending all the events for all the projects they have? 

If we do this I am not sure if we can set up trigger options separately for each project.

Reply:
> As you can see https://github.com/shreyaspapi/research/raw/master/gitlab_integration/gitlab-packit.png Users have trigger options for Push, Issue, Confidential Issue, Merge Request. Only if the option is ticked the events will be sent to us.

I am now confused about what values are in the settings-page by default and what can be set by us. So the example in the image shows the default values and we can add custom ones?

Reply:
> Even if we set up webhook globally for gnome will they be comfortable for sending all the events for all the projects they have?

Yes, that's a valid concern. We can have some opt-in value in the settings, but in that case, we still need to process all those events.

Reply:
> I am now confused about what values are in the settings-page by default and what can be set by us. So the example in the image shows the default values and we can add custom ones?

So just to make everything clear, There are no default values when adding project services! We can add or remove any type of parameters we want (this is where we can have all the config settings we thought of).

We just have to find alternative to this -

Even after installing to project, the service has no permissions to the repository. So for sending out the commit-status (Build information), a user will still have to give permissions to **Packit Gitlab User**